module-services/service-desktop/endpoints/filesystem/FS_Helper.cpp

```diff
@@ -1,23 +1,52 @@
-// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// Copyright (c) 2017-2023, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
 #include <endpoints/filesystem/FS_Helper.hpp>
-#include <endpoints/Context.hpp>
-#include <service-desktop/DesktopMessages.hpp>
 #include <service-desktop/ServiceDesktop.hpp>
-#include <endpoints/JsonKeyNames.hpp>
 #include <endpoints/message/Sender.hpp>
-#include <purefs/filesystem_paths.hpp>
 
 #include <sys/statvfs.h>
-#include <filesystem>
 
 namespace sdesktop::endpoints
 {
     namespace
     {
         constexpr auto bytesInMebibyte = 1024LLU * 1024LLU;
-    }
+
+        enum FileType
+        {
+            directory,
+            regularFile,
+            symlink,
+            other,
+        };
+
+        auto parseFileEntry(const std::filesystem::directory_entry &entry) -> json11::Json
+        {
+            int size = 0;
```

Comment:
`std::size_t` or in this case `uintmax_t` would be better in my opinion.

Reply:
yes, but JSON (our library) does not support anything larger than int, so it will be either way narrowed down to int